pageserver/src/walingest.rs

```diff
@@ -400,14 +401,26 @@ impl<'a> WalIngest<'a> {
             && decoded.xl_rmid == pg_constants::RM_XLOG_ID
             && (decoded.xl_info == pg_constants::XLOG_FPI
                 || decoded.xl_info == pg_constants::XLOG_FPI_FOR_HINT)
-        // compression of WAL is not yet supported: fall back to storing the original WAL record
-            && !postgres_ffi::bkpimage_is_compressed(blk.bimg_info, self.timeline.pg_version)?
+        // only lz4 compression of WAL is now supported, for other cmpression algorithms fall back to storing the original WAL record
+            && (!postgres_ffi::bkpimage_is_compressed(blk.bimg_info, self.timeline.pg_version)? ||
+                postgres_ffi::bkpimage_is_compressed_lz4(blk.bimg_info, self.timeline.pg_version)?)
```

Comment:
What do these booleans even mean?

```
            && {
                let is_compressed = postgres_ffi::bkpimage_is_compressed(blk.bimg_info, self.timeline.pg_version)?;
                let is_lz4_compressed = postgres_ffi::bkpimage_is_compressed_lz4(blk.bimg_info, self.timeline.pg_version)?;
                (!is_compressed || is_lz4_compressed)
            }
```

The first part looks weird to me.

What is the fallback for compressed with something else than lz4, and why do we decompress the lz4 here? Is that for walredo which cannot handle lz4 compressed images?